server/src/engine/flix.ts

```diff
@@ -202,15 +201,48 @@ export async function stop() {
   }
 }
 
+/**
+ * Add the given `uri` to the workspace.
+ */
 export function addUri(uri: string) {
+  currentWorkspaceFiles.add(uri)
+
   const job: jobs.Job = {
     request: jobs.Request.apiAddUri,
     uri,
   }
   queue.enqueue(job)
 }
 
+/**
+ * Handle a change in the file with the given `uri`.
+ *
+ * If this URI has not already been added to the workspace via {@linkcode addUri},
+ * it will be ignored, making it safe to call this function on any file.
+ */
+export function updateUri(uri: string, src: string) {
+  if (!currentWorkspaceFiles.has(uri)) {
```

Comment:
In the old code the request is sent even if the file is already added... because it has been updated.

Does this work correctly if you change a document? (without saving)

Reply:
This is the part that prevents irrelevant files from being added. 
`currentWorkspaceFiles` represents all files that the client has added, i.e. all files within the glob.

Reply:
Oh I see. I think I was tired.

Reply:
Ill try it tomorrow.

Reply:
Maybe add a comment: If the file is not in currentWorkSpaceFiles then it means it is outside the workspace, i.e. should be ignored.